indico/modules/events/timetable/legacy.py

```diff
@@ -36,6 +39,11 @@ def __init__(self, management=False):
         self.management = management
 
     def serialize_timetable(self, event, days=None, hide_weekends=False, strip_empty_days=False):
+        # pre-fetch ContributionPrincipals (to save time)
+        principal_map = {contrib.id: contrib.acl_entries for contrib in Contribution.find(
```

Comment:
I'd use `contrib` as the cache key. That way you could also include sessions in the cache without needing separate cache entries